File: examples/testeditor.c
#include <gtk/gtk.h>
#include "gtkmdeditor.h"

typedef struct {
    char *file_path;
} AppData;

static void activate(GtkApplication *app, gpointer user_data) {
    AppData *app_data = (AppData*)user_data;
    
    GtkWidget *window = gtk_application_window_new(app);
    gtk_window_set_title(GTK_WINDOW(window), "Markdown Editor Test");
    gtk_window_set_default_size(GTK_WINDOW(window), 600, 400);

    // Read content from file
    GError *error = NULL;
    gchar *content;
    if (!g_file_get_contents(app_data->file_path, &content, NULL, &error)) {
        g_printerr("Error reading file '%s': %s\n", app_data->file_path, error->message);
        g_error_free(error);
        g_application_quit(G_APPLICATION(app));
        return;
    }

    // Create a scrolled window
    GtkWidget *scrolled = gtk_scrolled_window_new();
    gtk_window_set_child(GTK_WINDOW(window), scrolled);

    // Add the editor to the scrolled window
    GtkWidget *editor = gtk_md_editor_new();
    GtkTextBuffer *text_buffer = gtk_text_view_get_buffer((GtkTextView*) editor);
    gtk_text_buffer_set_text(text_buffer, content, -1);
    g_free(content);
    
    gtk_scrolled_window_set_child(GTK_SCROLLED_WINDOW(scrolled), editor);

    gtk_window_present(GTK_WINDOW(window));
}

int main(int argc, char **argv) {
    AppData app_data;
    
    // Use provided file path or default to examples/test.md
    if (argc > 1) {
        app_data.file_path = argv[1];
    } else {
        app_data.file_path = "examples/test.md";
    }

    GtkApplication *app = gtk_application_new("org.gtk.example", G_APPLICATION_DEFAULT_FLAGS);
    g_signal_connect(app, "activate", G_CALLBACK(activate), &app_data);
    int status = g_application_run(G_APPLICATION(app), argc, argv);
    g_object_unref(app);

    return status;
}

File: examples/testeditor.rs
use gtk::glib;
use gtk::prelude::*;
use gtk4 as gtk;
use std::fs;
use std::env;

fn main() -> glib::ExitCode {
    let application = gtk::Application::builder()
        .application_id("com.bjareholt.johan.GtkMdEditor")
        .build();

    application.connect_activate(|app| {
        let window = gtk::ApplicationWindow::builder()
            .application(app)
            .title("GTK Markdown Editor Test")
            .default_width(800)
            .default_height(600)
            .build();

        // Create scrolled window
        let scroll = gtk::ScrolledWindow::builder()
            .hexpand(true)
            .vexpand(true)
            .build();

        let mdeditor = gtkmdeditor::GtkMdEditor::new();

        // Get file path from args or use default
        let args: Vec<String> = env::args().collect();
        let file_path = if args.len() > 1 {
            &args[1]
        } else {
            "examples/test.md"
        };

        // Read markdown from file
        let markdown_content = match fs::read_to_string(file_path) {
            Ok(content) => content,
            Err(e) => {
                eprintln!("Error reading file '{}': {}", file_path, e);
                format!("# Error\n\nCould not read file: {}\n\nError: {}", file_path, e)
            }
        };

        // Set the content in the editor
        let buffer = mdeditor.buffer();
        buffer.set_text(&markdown_content);

        scroll.set_child(Some(&mdeditor));
        window.set_child(Some(&scroll));

        window.present();
    });

    application.run_with_args::<&String>(&[])
}
